### PR TITLE
Mzlib Update to 1.0.542 - Averaging crash fix+

### PR DESCRIPTION
**Pull request by nbollis**
Update to most recent version of mzlib which fixes a crash in SpectralAveraging. 

Mzlib PR is found at https://github.com/smith-chem-wisc/mzLib/pull/739

Cleaned up GUI switch cases. 
Eliminated AverageDdaScansWithOverlap and updated toml format to automaticaly convert old toml's with AverageDdaScansWithOverlap to load in as AverageDdaScans

### CODE REVIEW

File: MetaMorpheus/GuiFunctions/MetaDraw/MetaDrawLogic.cs [hunk 1]
```diff
@@ -795,6 +796,7 @@ public void CleanUpSpectraFiles()
                 {
                     connection.Value.CloseDynamicConnection();
                 }
+                Thread.Sleep(1000);
```
Comment (trishorts) [inline MetaMorpheus/GuiFunctions/MetaDraw/MetaDrawLogic.cs:799]: let's put a comment here explaining what happens if you don't have this sleep.

**Review by trishorts — APPROVED** [review]

File: MetaMorpheus/GuiFunctions/MetaDraw/MetaDrawLogic.cs (hunk 1 above)
Comment (Alexander-Sol) [reply MetaMorpheus/GuiFunctions/MetaDraw/MetaDrawLogic.cs:799]: Seconded

**Review by Alexander-Sol — APPROVED** [review]
When I ran the averaging task, it wrote a blank .toml file in the Task1-AveragingTask folder. 
Other than that, everything works great.

**Comment by trishorts** [comment]
I feel like we better fix the empty toml thing

**Comment by nbollis** [comment]
> When I ran the averaging task, it wrote a blank .toml file in the Task1-AveragingTask folder. Other than that, everything works great

> I feel like we better fix the empty toml thing

Took care of this. It will now only spit out the toml if file specific parameters (most likely from a preceeding calibration task) are present for the averaged spectra file

File: MetaMorpheus/GuiFunctions/MetaDraw/MetaDrawLogic.cs (hunk 1 above)
Comment (nbollis) [reply MetaMorpheus/GuiFunctions/MetaDraw/MetaDrawLogic.cs]: Comment added

**Pull request merged**